### PR TITLE
python3Packages.pywiim: init at 2.3.7; music-assistant: add pywiim to wiim provider; music-assistant: add tebriel as maintainer

### PR DESCRIPTION
**Pull request by tebriel**
- python3Packages.pywiim: init at 2.3.7
- music-assistant: include pywiim for wiim provider
- music-assistant: add tebriel as maintainer

In https://github.com/music-assistant/server/pull/5729 the WiiM provider now fails to load if `pywiim` isn't present. This manifests as an error in the interface saying that the wiim provider is not installed. But really it's just that wiim provider needs `pywiim`. I've initialized `pywiim`, added it to the wiim provider, and added myself as a maintainer as I saw that @emilylange is looking to possibly [come off the maintainer list soon](https://github.com/NixOS/nixpkgs/pull/547702#pullrequestreview-5211888097), so I'm volunteering to help out.

<details>
<summary>Example Stacktrace/Error Logs</summary>

```
Sep 19 10:39:29 bard mass[263413]: 2026-09-19 10:39:29.055 WARNING (MainThread) [music_assistant] Error loading provider(instance) wiim: Configure wiim in `services.music-assistant.providers` to install the required dependencies.
Sep 19 10:39:29 bard mass[263413]: Traceback (most recent call last):
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/helpers/util.py", line 1425, in load_provider_module
Sep 19 10:39:29 bard mass[263413]:     return await _get_provider_module()
Sep 19 10:39:29 bard mass[263413]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/helpers/util.py", line 1420, in _get_provider_module
Sep 19 10:39:29 bard mass[263413]:     module = await import_module_in_thread(f".{domain}", "music_assistant.providers")
Sep 19 10:39:29 bard mass[263413]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/helpers/util.py", line 1405, in import_module_in_thread
Sep 19 10:39:29 bard mass[263413]:     return await loop.run_in_executor(_IMPORT_EXECUTOR, importlib.import_module, name, package)
Sep 19 10:39:29 bard mass[263413]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/d64q19q1xjdwfhqx6czvrjgrhq0n3lcc-python3-3.14.7/lib/python3.14/concurrent/futures/thread.py", line 86, in run
Sep 19 10:39:29 bard mass[263413]:     result = ctx.run(self.task)
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/d64q19q1xjdwfhqx6czvrjgrhq0n3lcc-python3-3.14.7/lib/python3.14/concurrent/futures/thread.py", line 73, in run
Sep 19 10:39:29 bard mass[263413]:     return fn(*args, **kwargs)
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/d64q19q1xjdwfhqx6czvrjgrhq0n3lcc-python3-3.14.7/lib/python3.14/importlib/__init__.py", line 88, in import_module
Sep 19 10:39:29 bard mass[263413]:     return _bootstrap._gcd_import(name[level:], package, level)
Sep 19 10:39:29 bard mass[263413]:            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 10:39:29 bard mass[263413]:   File "<frozen importlib._bootstrap>", line 1406, in _gcd_import
Sep 19 10:39:29 bard mass[263413]:   File "<frozen importlib._bootstrap>", line 1371, in _find_and_load
Sep 19 10:39:29 bard mass[263413]:   File "<frozen importlib._bootstrap>", line 1342, in _find_and_load_unlocked
Sep 19 10:39:29 bard mass[263413]:   File "<frozen importlib._bootstrap>", line 938, in _load_unlocked
Sep 19 10:39:29 bard mass[263413]:   File "<frozen importlib._bootstrap_external>", line 759, in exec_module
Sep 19 10:39:29 bard mass[263413]:   File "<frozen importlib._bootstrap>", line 491, in _call_with_frames_removed
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/providers/wiim/__init__.py", line 13, in <module>
Sep 19 10:39:29 bard mass[263413]:     from .provider import WiimProvider
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/providers/wiim/provider.py", line 12, in <module>
Sep 19 10:39:29 bard mass[263413]:     from pywiim import WiiMClient, WiiMError
Sep 19 10:39:29 bard mass[263413]: ModuleNotFoundError: No module named 'pywiim'
Sep 19 10:39:29 bard mass[263413]: During handling of the above exception, another exception occurred:
Sep 19 10:39:29 bard mass[263413]: Traceback (most recent call last):
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/mass.py", line 989, in load_provider
Sep 19 10:39:29 bard mass[263413]:     await self.load_provider_config(prov_conf)
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/mass.py", line 909, in load_provider_config
Sep 19 10:39:29 bard mass[263413]:     await self._load_provider(prov_conf)
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/mass.py", line 1355, in _load_provider
Sep 19 10:39:29 bard mass[263413]:     prov_mod = await load_provider_module(domain, prov_manifest.requirements)
Sep 19 10:39:29 bard mass[263413]:                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 19 10:39:29 bard mass[263413]:   File "/nix/store/0wi2f1x34i0nan0ckc7lhi39pn18lrk1-music-assistant-2.10.3/lib/python3.14/site-packages/music_assistant/helpers/util.py", line 1427, in load_provider_module
Sep 19 10:39:29 bard mass[263413]:     raise RuntimeError(f"Configure {domain} in `services.music-assistant.providers` to install the required dependencies.")
Sep 19 10:39:29 bard mass[263413]: RuntimeError: Configure wiim in `services.music-assistant.providers` to install the required dependencies.
```

</details>

## Things done

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.
- [x] Follows the [automation/AI policy].

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[automation/AI policy]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

### CODE REVIEW

**Comment by Sigmanificient** [comment]
While i love using `@` as a replacement for at, i think we should not add more edge-cases to our commit convention :sweat_smile:

**Comment by tebriel** [comment]
@Sigmanificient ha gotcha! Sorry I built that string off of memory I thought it was @ I'll go refresh myself on the commit conventions. Updated the commit.

File: pkgs/development/python-modules/pywiim/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  nativeCheckInputs = [
+    pytestCheckHook
+
+    # dev dependencies
+    black
+    isort
+    ruff
+    mypy
+    pytest
```
Comment (mweinelt) [inline pkgs/development/python-modules/pywiim/default.nix:60]: none of these make sense

File: pkgs/development/python-modules/pywiim/default.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    pytest-asyncio
+    pytest-cov
```
Comment (mweinelt) [inline pkgs/development/python-modules/pywiim/default.nix]:
```suggestion
    pytest-cov-stub
```

File: pkgs/development/python-modules/pywiim/default.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  ++ optional-dependencies.mcp;
+
+  pythonImportsCheck = [ "pywiim" ];
+
+  meta = with lib; {
+    description = "Python library for WiiM/LinkPlay device communication";
+    homepage = "https://github.com/mjcumming/pywiim";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = with lib.maintainers; [ tebriel ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/pywiim/default.nix:76]:
```suggestion
    maintainers = with lib.maintainers; [ tebriel ];
    mainProgram = "....";
```

One of these?

https://github.com/mjcumming/pywiim/blob/main/pyproject.toml#L60C1-L67C28

File: pkgs/development/python-modules/pywiim/default.nix [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    aiohttp
+    pydantic
+    async-upnp-client
+    m3u8
+    mutagen
```
Comment (mweinelt) [inline pkgs/development/python-modules/pywiim/default.nix:45]: sort

File: pkgs/development/python-modules/pywiim/default.nix [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  # dev dependencies
+  black,
+  isort,
+  ruff,
+  mypy,
+  pytest,
+  pytest-asyncio,
+  pytest-cov,
+  pytest-xdist,
+  pyyaml,
```
Comment (mweinelt) [inline pkgs/development/python-modules/pywiim/default.nix:20]: prune

File: pkgs/development/python-modules/pywiim/default.nix [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+  # mcp deps
+  mcp,
```
Comment (mweinelt) [inline pkgs/development/python-modules/pywiim/default.nix]:
```suggestion
  # optional deps
  mcp,
```

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pywiim/default.nix [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+}:
+buildPythonPackage rec {
+  pname = "pywiim";
+  version = "2.3.7";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "mjcumming";
+    repo = "pywiim";
+    rev = "v${version}";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pywiim/default.nix]:
```suggestion
    tag = "v${finalAttrs.version}";
```

File: pkgs/development/python-modules/pywiim/default.nix [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    pytest-cov
+    pytest-xdist
+    pyyaml
+  ]
+  ++ optional-dependencies.mcp;
+
+  pythonImportsCheck = [ "pywiim" ];
+
+  meta = with lib; {
+    description = "Python library for WiiM/LinkPlay device communication";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pywiim/default.nix:72]: https://github.com/mjcumming/pywiim/blob/v2.3.7/CHANGELOG.md
